test/cpp/qps/client.h

```diff
@@ -280,19 +274,21 @@ class Client {
         if (entry.used()) {
           histogram_.Add(entry.value());
         }
+        bool done = false;
         if (!thread_still_ok) {
           gpr_log(GPR_ERROR, "Finishing client thread due to RPC error");
-          done_ = true;
+          done = true;
         }
-        if (done_) {
+        done = done || (gpr_atm_acq_load(&client_->thread_pool_done_) !=
+                        static_cast<gpr_atm>(0));
+        if (done) {
```

Comment:
you can do away with `done` by doing:

```
if (!thread_still_ok) {
  gpr_log(...);
}
if (!thread_still_ok || (gpr_atm_acq_load(&client_->thread_pool_done_) != static_cast<gpr_atm>(0)) {
  client_->CompleteThread();
  return;
}
```

Arguably, that may hurt readability, so I don't feel strongly about it.